lib/coffee_place/cli_options.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module CoffeePlace
+  # Keeps parsed CLI options, also defining defaults.
+  class CLIOptions
+    attr_accessor :version, :help, :remaining, :num_results
+
+    def initialize(version: false, help: false, remaining: [])
```

Comment:
Interesting use of flags. Generally, are there any issues that could arise from the use of boolean params?

Reply:
Accidentally passing in `nil` is always a major pain.
There is always the possibility of someone passing a `String` too - such as `'false'`.

`ActionControlller` usually does some light type casting to ensure values sent from checkboxes
or other fields are converted into boolean values.

When I use these sort of params internally in my code, I try to make sure that I always send a boolean,
even when the value is missing - by passing in a default value.

When using user submitted fields, some sort of type casting is useful.
I have found this [StackOverflow answer](https://stackoverflow.com/questions/36228873/ruby-how-to-convert-a-string-to-boolean#answer-44322375) helpful when parsing user submitted CSV files,
that contain columns which represent boolean values.

Reply:
This are all valid points, but there is also a code smell that arises from using bools in your methods, you can read more about it in this [Martin Fowler article, dating way back in 2011](https://martinfowler.com/bliki/FlagArgument.html)

Reply:
One more article recommendation from me: [Clean code: The curse of a boolean parameter](https://medium.com/@amlcurran/clean-code-the-curse-of-a-boolean-parameter-c237a830b7a3).

Reply:
I agree that using boolean arguments for control flow leads to needless complexity.
Having simpler methods to handle each case is better.
Thank you for the two articles.